pkg/controlplane/instance.go

```diff
@@ -184,10 +185,9 @@ func (cp *Instance) CreateExport(export *cpstore.Export) error {
 		if err := cp.exports.Create(export); err != nil {
 			return err
 		}
-		// create k8s endpoint and service for external service.
-		if exSvc.Host != "" && exSvc.Port != 0 && cp.initialized {
-			cp.platform.CreateEndpoint(export.Name, exSvc.Host, exSvc.Port)
-			cp.platform.CreateService(export.Name, export.Name, exSvc.Port, exSvc.Port)
+		// create a k8s external service.
+		if eSpec.ExternalService != "" {
+			cp.platform.CreateExternalService(exportPrefix+export.Name, eSpec.Service.Host, eSpec.ExternalService, eSpec.Service.Port)
```

Comment:
See earlier comment regarding the port - not sure it is needed.
For example, there is no use for port in `setExternalNameService`.

Reply:
Yes the port it is just for the print, I can remove it